### ISSUE TITLE
JNI error when trying to run mqtt5 sample on android

### ISSUE THREAD

**Issue by Suzuhana** [aws/aws-iot-device-sdk-java-v2#365]
### Describe the bug

I am getting below error when I trying to run the codes from the mqtt5 sample:
```
java_vm_ext.cc:570] JNI DETECTED ERROR IN APPLICATION: the return type of CallObjectMethod does not match void software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions$LifecycleEvents.onAttemptingConnect(software.amazon.awssdk.crt.mqtt5.Mqtt5Client, software.amazon.awssdk.crt.mqtt5.OnAttemptingConnectReturn)
java_vm_ext.cc:570]     in call to CallObjectMethod
```

My current dependencies are below: 

```
    // https://mvnrepository.com/artifact/software.amazon.awssdk.crt/aws-crt-android
    implementation group: 'software.amazon.awssdk.crt', name: 'aws-crt-android', version: '0.21.6'
    implementation ('software.amazon.awssdk.iotdevicesdk:aws-iot-device-sdk:1.11.4') {
        exclude group: 'software.amazon.awssdk.crt', module: 'aws-crt'
    }

```

Same code is able to run on My mac.

Additionally, the example for running mqtt over websocket works fine on android (with same dependencies above).

### Expected Behavior

I expect the example should able to run in android.

### Current Behavior

It is crashing in android when 

```
lifecycleEvents.connectedFuture.get(60, TimeUnit.SECONDS);
```
is called.

### Reproduction Steps

Use dependencies as :
```
    // https://mvnrepository.com/artifact/software.amazon.awssdk.crt/aws-crt-android
    implementation group: 'software.amazon.awssdk.crt', name: 'aws-crt-android', version: '0.21.6'
    implementation ('software.amazon.awssdk.iotdevicesdk:aws-iot-device-sdk:1.11.4') {
        exclude group: 'software.amazon.awssdk.crt', module: 'aws-crt'
    }
```
Copy  Mqtt5 example codes into a sample android app and supply necessary arguments.

Run and it crashes.

### SDK version used

1.11.4   and aws-crt-android 0.21.6

### Environment details (OS name and version, etc.)

Android 10

**Comment by bretambrose** [comment]
It looks like we're using incorrect JNI methods for the lifecycle events (should be CallVoidMethod rather than CallObjectMethod).  I'll update the CRT and do a quick check of the other native-to-Java invocations as well.  Will update when a release has been made.

**Comment by Suzuhana** [comment]
The issue is resolved in 0.21.7
